compiler+runtime/src/cpp/jank/runtime/obj/native_pointer_wrapper.cpp

```diff
@@ -0,0 +1,45 @@
+#include <jank/runtime/obj/native_pointer_wrapper.hpp>
+
+namespace jank::runtime
+{
+  obj::native_pointer_wrapper::static_object(void *d)
+    : data{ d }
+  {
+  }
+
+  native_bool obj::native_pointer_wrapper::equal(object const &o) const
+  {
+    if(o.type != object_type::native_pointer_wrapper)
+    {
+      return false;
+    }
+
+    auto const c(expect_object<obj::native_pointer_wrapper>(&o));
```

Comment:
Please use `dyn_cast` instead of the pattern of checking for the type and then using `expect_type`.

```cpp
if(auto const c = dyn_cast<obj::native_pointer_wrapper>(&o))
{
  return data == c->data;
}
```